### PR TITLE
Added detection of Azure Pipelines.

### PR DESCRIPTION
**Pull request by dlemstra**
This PR adds detection of Azure Pipelines to the script. It might need some tweaks but this seems to work for me now.

### CODE REVIEW

**Comment by jfboismenu** [comment]
Looking forward to this making it to the master branch.

**Comment by astrofrog** [comment]
Looking forward to this!

**Comment by felixfbecker** [comment]
This would be great. Currently using this as a workaround:
```yml
      - bash: bash <(curl -s https://codecov.io/bash) -f coverage.json -P "$SYSTEM_PULLREQUEST_PULLREQUESTNUMBER"
        env:
          CI_BUILD_ID: $(Build.BuildNumber)-$(Agent.OS)
          CODECOV_NAME: ${{ parameters.name }}
          VCS_BRANCH_NAME: $(Build.SourceBranchName)
          VCS_COMMIT_ID: $(Build.SourceVersion)
          CODECOV_TOKEN: $(CodeCovToken)
        displayName: Upload coverage to CodeCov
```

**Comment by blueyed** [comment]
According to https://github.com/codecov/codecov-bash/issues/171 this appears what you get currently already from https://codecov.io/bash ?!

(and it breaks my manual setting of `VCS_BRANCH_NAME` then.. - via https://github.com/pytest-dev/pytest/pull/5403)

File: codecov [hunk 1]
```diff
@@ -735,6 +735,22 @@ then
...
+  commit="$BUILD_SOURCEVERSION"
+  build="$BUILD_BUILDNUMBER"
+  if [ "$PULL_REQUEST_NUMBER" != "$PULL_REQUEST_ID" ];
+  then
+    pr="$PULL_REQUEST_NUMBER"
+  else
+    pr="$PULL_REQUEST_ID"
+  fi
+  job="${BUILD_BUILDID}"
+  branch="$BUILD_SOURCEBRANCHNAME"
```
Comment (blueyed) [inline codecov]: This needs to be different for pull requests (`$SYSTEM_PULLREQUEST_TARGETBRANCH`) - where it currently ends up in "merge".  But maybe that's OK and only something that looks weird?

**Comment by astrofrog** [comment]
@dlemstra - see my comment in https://github.com/codecov/codecov-bash/issues/189 about some of the variables being incorrect - this appears to be correct in codecov.exe but not here.

**Pull request closed without merging**